src/sentry/migrations/0150_remove_slack_workspace_orgintegrations.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.29 on 2021-01-20 19:29
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db.models import Q
+
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def remove_slack_workspace_apps(apps, schema_editor):
+    """
+    Remove OrganizationIntegrations that are linked to Slack Integrations that are
+    still using workspace apps.
+
+    We can determine which integrations are workspace apps by looking at the metadata.
+    There is a field for new or migrated slack integrations called the `installation_type`.
+    If a slack integration is still a workspace app, it will not have this field
+    in the metadata.
+    """
+
+    Integration = apps.get_model("sentry", "Integration")
+    OrganizationIntegration = apps.get_model("sentry", "OrganizationIntegration")
+
+    integrations = (
+        Integration.objects.all()
+        .filter(provider="slack")
+        .exclude(metadata__icontains="installation_type")
+        .values_list("id", flat=True)
+    )
```

Comment:
There are 32k slack integrations here. When looking at `OrganizationIntegration`, there are only at most 23 rows per integration. I'm not sure that postgres is going to be super happy with having a list of 10s of thousands of ids passed to it.

My recommendation would be to restructure this so that you use `RangeQuerySetWrapperWithProgressBar` to iterate over `Integration` instead. Then for each integration, do 
```
for org_integration in OrganizationIntegration.objects.filter(integration_id=integration_id):
    org_integration.delete()
```